src/include/migraphx/matcher.hpp

```diff
@@ -896,10 +897,20 @@ inline auto has_value(T x, std::size_t atol_mult = 10, std::size_t rtol_mult = 1
             else
             {
                 auto eps = std::numeric_limits<type>::epsilon();
-                if(std::all_of(v.begin(), v.end(), [&](auto val) {
-                       return std::fabs(val - static_cast<type>(x)) <
-                              eps * (atol_mult + rtol_mult * std::fabs(val));
-                   }))
+                if(atol_mult == 0 and rtol_mult == 0)
+                {
+                    // do exact match using float_equal if atol and rtol both are set to zero
+                    if(std::all_of(v.begin(), v.end(), [&](auto val) {
+                           return migraphx::float_equal(std::fabs(val), static_cast<type>(x));
```

Comment:
Question: wouldn't it be logically correct, and more efficient, to do float_equal() using val -- instead of using std::fabs(val) ?

Reply:
Can you add a test for `has_value(5.0, 0, 0)` that checks against `-5.0`(which should fail to match)?

Reply:
> Can you add a test for has_value(5.0, 0, 0) that checks against -5.0(which should fail to match)? 

added

> ) using val -- instead of using std::fabs(val) ? 

Yes fixed that.